server/proposer_pk_storage.go

```diff
@@ -0,0 +1,9 @@
+package server
+
+import "github.com/flashbots/go-boost-utils/types"
+
+// proposerPublicKeyStorage is used to store a proposer's public key according to a specific slot
+// along with the block hash it is currently proposing.
+type proposerPublicKeyStorage struct {
+	storage map[string]types.PublicKey
```

Comment:
instead of a concatenated string as key, maybe use a struct with explicit slot and hash.

also a helper method on this to expire entries older than a slot or two might be useful.

Reply:
Added in 7f917d8ef552501817ba72ec602baea5fe16e6c4 and 63f5716a80628e7e28662829651241468ef3543a.